### PR TITLE
WIP: Updating IGListGridCollectionViewLayout to be more flexible

### LINKED ISSUE #263: Center aligned flow layout
Hi thanks for this amazing library! loving it so far, the only issue that i have right now is that i need a center aligned collection view layout, is there any way we could add some sort of alignment to the new `IGListGridCollectionViewLayout` so that we could have both left and center alignment?

### PR DESCRIPTION
**Pull request by zhubofei**
## Changes in this pull request
Updating the grid layout to fit more general cases. 

Closes #263

## TODO 
- [x] Add support for multi-item sections.
- [x] Add unit test for multi-item sections.
- [x] Add `IGListGridCollectionViewLayoutAlignment` option. Allow user to set alignment for the layout.
- [x] Add unit test for alignments.
- [ ] Add support for horizontal scrolling.
- [ ] Add unit test for horizontal mode.
- [ ] Add support for section inset.
- [ ] Add unit test for section inset.
- [ ] Add attributes cache.
- [ ] Update Changelog

### CODE REVIEW

File: Source/IGListGridCollectionViewLayout.h [hunk 1]
```diff
@@ -10,6 +10,16 @@
 #import <UIKit/UIKit.h>
 #import <IGListKit/IGListMacros.h>
 
+/**
+ An option for how to align items in grid layout.
+ */
+typedef NS_ENUM(NSInteger, IGListGridCollectionViewLayoutAlignment) {
+    IGListGridCollectionViewLayoutAlignDefault,
```
Comment (jessesquires) [inline Source/IGListGridCollectionViewLayout.h]: - what does `Default` mean? i would expect to see `Left`, `Right`, `Center`... 
- nit: can we spell-out `Alignment` for each enum case? (follows conventions)

**Comment by rnystrom** [comment]
@zhubofei heads up that we are talking internally about adding a layout we built for IG to IGLK. It does all of this and is wicked fast. If we do, we'll probably decide to stick w/ that one layout. Just want you to know so you don't spend tons of time on this just for us to replace it.

Sorry if this causes some thrash!

**Comment by zhubofei** [comment]
@rnystrom This is a great news! I'm actually really worried about the performance of my implementation and cannot come up with a better layout algorithm. Really curious about the wicked fast algorithm from you guys, can't wait to see it!

**Comment by zhubofei** [comment]
@rnystrom The new layout is superb! I actually thought about rewriting the whole thing with objective-c++ before, but lack the motivation to do so 😂. 

I'll close this one then.

**Comment by ShaneQi** [comment]
@zhubofei Since [pr#484](https://github.com/Instagram/IGListKit/pull/484) and [pr#450](https://github.com/Instagram/IGListKit/pull/450) are already closed, do you think we can expect this pull request reopen?

Anyway, I have some comments:
- Alignment layout is very useful.
- Is it possible that different sections have different `IGListGridCollectionViewLayoutAlignment` options? (sounds like tons of work to implement this)

Thanks!

**Comment by zhubofei** [comment]
@ShaneQi We need @rnystrom 's opinion on the alignment option. I'm happy to work on it if he approves this option.

> Is it possible that different sections have different IGListGridCollectionViewLayoutAlignment options? (sounds like tons of work to implement this)

This one, though, is definitely too complicated for a general layout. I suggest you subclass the layout and customize it as needed.

**Comment by ShaneQi** [comment]
Thanks for the reply and all the contribution you've done!

**Comment by rnystrom** [comment]
@ShaneQi sadly we replaced this layout with #484. Agree w/ @zhubofei that alignment might just be beyond the scope of IGListKit and require a more comprehensive `UICollectionViewLayout` entirely (there are some great libs out there already).

If someone has an elegant way to extend our layout, I'm very into a PR to try some stuff out.

**Comment by ShaneQi** [comment]
@rnystrom @zhubofei I have to say you are right, aligning cells in some specific sections is too specific. One should subclass `UICollectionViewLayout` to achieve that. 

BTW, thanks for bringing new layout to us.

**Pull request closed without merging**